FindBaselibs.cmake

```diff
@@ -92,10 +74,41 @@ if (PFUNIT_FOUND)
   add_custom_target(tests COMMAND ${CMAKE_CTEST_COMMAND})
 endif ()
 
-# BASEDIR.rc file does not have the arch
-string(REPLACE "/${CMAKE_SYSTEM_NAME}" "" BASEDIR_WITHOUT_ARCH ${BASEDIR})
-set(BASEDIR_WITHOUT_ARCH ${BASEDIR_WITHOUT_ARCH} CACHE STRING "BASEDIR without arch")
-mark_as_advanced(BASEDIR_WITHOUT_ARCH)
+if(BASEDIR)
+  # BASEDIR.rc file does not have the arch
+  string(REPLACE "/${CMAKE_SYSTEM_NAME}" "" BASEDIR_WITHOUT_ARCH ${BASEDIR})
+  set(BASEDIR_WITHOUT_ARCH ${BASEDIR_WITHOUT_ARCH} CACHE STRING "BASEDIR without arch")
+  mark_as_advanced(BASEDIR_WITHOUT_ARCH)
+endif()
 
 # Set the site variable
 include(DetermineSite)
+
+
+# Make Baselibs target
+add_library(Baselibs INTERFACE)
+target_include_directories(Baselibs INTERFACE ${NETCDF_INCLUDE_DIRS})
+target_link_libraries(Baselibs INTERFACE 
+    $<$<TARGET_EXISTS:gftl-shared>:gftl-shared>
+    $<$<TARGET_EXISTS:fargparse>:fargparse>
+    $<$<TARGET_EXISTS:FLAP>:FLAP>
+    $<$<TARGET_EXISTS:pfunit>:pfunit>
+    gftl ESMF
+    MPI::MPI_C MPI::MPI_CXX MPI::MPI_Fortran
+    OpenMP::OpenMP_Fortran
```

Comment:
I am very concerned about this. Does this mean all of GEOS will build with OpenMP?

We definitely **DO NOT** want OpenMP turned on by fiat. There is unexercised and scattered OpenMP code in the Physics that *should not* be turned on. We build with OpenMP very selectively for a good reason.

I just tried a Release CMake with this (and bits of the MAPL PR) and I see `-qopenmp` in the `flags.make` of a lot of Physics code, e.g.,
```
src/Components/@GEOSgcm_GridComp/GEOSagcm_GridComp/GEOSphysics_GridComp/@GEOSchem_GridComp/GOCART_GridComp/CFC_GridComp/CMakeFiles/CFC_GridComp.dir/flags.make
src/Components/@GEOSgcm_GridComp/GEOSagcm_GridComp/GEOSphysics_GridComp/@GEOSchem_GridComp/GOCART_GridComp/CO_GridComp/CMakeFiles/CO_GridComp.dir/flags.make
src/Components/@GEOSgcm_GridComp/GEOSagcm_GridComp/GEOSphysics_GridComp/@GEOSchem_GridComp/GOCART_GridComp/OC_GridComp/CMakeFiles/OC_GridComp.dir/flags.make
src/Components/@GEOSgcm_GridComp/GEOSagcm_GridComp/GEOSphysics_GridComp/@GEOSchem_GridComp/Shared/HEMCO/CMakeFiles/HEMCO.dir/flags.make
src/Components/@GEOSgcm_GridComp/GEOSagcm_GridComp/GEOSphysics_GridComp/@GEOSchem_GridComp/GOCART_GridComp/DU_GridComp/CMakeFiles/DU_GridComp.dir/flags.make
src/Components/@GEOSgcm_GridComp/GEOSagcm_GridComp/GEOSphysics_GridComp/@GEOSchem_GridComp/GOCART_GridComp/SU_GridComp/CMakeFiles/SU_GridComp.dir/flags.make
src/Components/@GEOSgcm_GridComp/GEOSagcm_GridComp/GEOSphysics_GridComp/@GEOSchem_GridComp/GOCART_GridComp/CO2_GridComp/CMakeFiles/CO2_GridComp.dir/flags.make
src/Components/@GEOSgcm_GridComp/GEOSagcm_GridComp/GEOSphysics_GridComp/@GEOSchem_GridComp/GOCART_GridComp/O3_GridComp/CMakeFiles/O3_GridComp.dir/flags.make
src/Components/@GEOSgcm_GridComp/GEOSagcm_GridComp/GEOSphysics_GridComp/@GEOSchem_GridComp/GOCART_GridComp/NI_GridComp/CMakeFiles/NI_GridComp.dir/flags.make
src/Components/@GEOSgcm_GridComp/GEOSagcm_GridComp/GEOSphysics_GridComp/@GEOSchem_GridComp/GOCART_GridComp/SS_GridComp/CMakeFiles/SS_GridComp.dir/flags.make
src/Components/@GEOSgcm_GridComp/GEOSagcm_GridComp/GEOSphysics_GridComp/@GEOSchem_GridComp/GOCART_GridComp/BRC_GridComp/CMakeFiles/BRC_GridComp.dir/flags.make
```

Reply:
@mathomp4 Thanks, this is good to know. Things like this are why I think it's really important GCHPctm can share GEOS-ESM's build system.

I have a list of changes to be made in the PR's thread. This can be remedied by explicitly linking only the dependencies each target needs (axing the Baselibs target all together).

Reply:
I removed the monolithic `Baselibs` target in 9ee1e28acf1851121dc89c478b29417d0d45d550. I added a `GEOS_BuildProperties` target with global compiler options and definitions but **no link libraries**. Instead, targets will have to link their own dependencies.